src/components/event/EventFormBase.vue

```diff
@@ -197,8 +205,110 @@ export default class EventFormBase extends Vue {
     }
   }
 
+  isChanged(): boolean {
+    if (this.summary && this.originalSummary) {
+      return (
+        JSON.stringify(this.summary) !== JSON.stringify(this.originalSummary)
+      )
+    }
+    return false
+  }
+
+  cleanupContent(): void {
+    this.summary.name = ''
+    this.summary.description = ''
+    this.summary.tags = []
+    this.summary.groupName = ''
+    this.summary.place = ''
+    this.summary.timeStart = ''
+    this.summary.timeEnd = ''
+    this.summary.open = false
+    this.summary.sharedRoom = true
+  }
+
+  isEventNewOrEdit(): boolean {
+    const currentRoute: Route = this.$route
+    return currentRoute.name === 'EventNew'
+  }
+
+  beforLeaveGuardinEventEdit = (to, from, next) => {
+    if (from.name === 'EventEdit') {
+      if (this.isChanged()) {
+        if (
+          confirm(
+            '入力されたデータは送信されないまま破棄されますが，よろしいですか。'
+          )
+        ) {
+          removeDraftConfirmer()
+          this.cleanupContent()
+          next()
+        } else {
+          next(false)
+        }
+      } else {
+        next()
+      }
+    } else {
+      next()
+    }
+  }
```

Comment:
こここんな感じでearly returnさせるとネストが浅くなって読みやすくなるかも
```suggestion
  beforLeaveGuardinEventEdit = (to, from, next) => {
    if (from.name !== 'EventEdit' || this.isChanged()) {
      next()
    }
    if (
      confirm(
        '入力されたデータは送信されないまま破棄されますが，よろしいですか。'
      )
    ) {
      removeDraftConfirmer()
      this.cleanupContent()
      next()
    }
    next(false)
  }
```